sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java

```diff
@@ -0,0 +1,295 @@
+        for (CosmosAsyncClient asyncClient : clientDocsMap.keySet()) {
+            asyncClient.close();
+        }
+    }
+
+    protected void onSuccess() {
+    }
+
+    protected void onError(Throwable throwable) {
+    }
+
+    private PojoizedJson generateDocument(String idString, String dataFieldValue, String partitionKey) {
+        com.azure.cosmos.benchmark.PojoizedJson instance = new com.azure.cosmos.benchmark.PojoizedJson();
+        Map<String, String> properties = instance.getInstance();
+        properties.put("id", idString);
+        properties.put(partitionKey, idString);
+
+        for (int i = 0; i < configuration.getDocumentDataFieldCount(); i++) {
+            properties.put("dataField" + i, dataFieldValue);
+        }
+
+        return instance;
+    }
+
+    private boolean shouldContinue(long startTimeMillis, long iterationCount) {
+
+        Duration maxDurationTime = configuration.getMaxRunningTimeDuration();
+        int maxNumberOfOperations = configuration.getNumberOfOperations();
+
+        if (maxDurationTime == null) {
+            return iterationCount < maxNumberOfOperations;
+        }
+
+        if (startTimeMillis + maxDurationTime.toMillis() < System.currentTimeMillis()) {
+            return false;
+        }
+
+        if (maxNumberOfOperations < 0) {
+            return true;
+        }
+
+        return iterationCount < maxNumberOfOperations;
+    }
+
+    private void createClients() {
+        String csvFile = "clientHostAndKey.txt";
+        String line = "";
+        String splitBy = ";";
+        try (BufferedReader br = new BufferedReader(new FileReader(csvFile))) {
+            while ((line = br.readLine()) != null) {
+                String[] hostAndKey = line.split(splitBy);
+                if (hostAndKey.length >= 2) {
+                    String endpoint = hostAndKey[0].substring(hostAndKey[0].indexOf(ACCOUNT_ENDPOINT_TAG) + ACCOUNT_ENDPOINT_TAG.length());
+                    String key = hostAndKey[1].substring(hostAndKey[1].indexOf(ACCOUNT_KEY_TAG) + ACCOUNT_KEY_TAG.length());
+                    CosmosAsyncClient asyncClient = new CosmosClientBuilder()
+                        .endpoint(endpoint)
+                        .key(key)
+                        .connectionPolicy(configuration.getConnectionPolicy())
+                        .consistencyLevel(configuration.getConsistencyLevel())
+                        .connectionReuseAcrossClientsEnabled(true)
+                        .buildAsyncClient();
+                    List<PojoizedJson> docsToRead = new ArrayList<>();
+                    if (!configuration.isDeleteCollections()) {
+                        CosmosAsyncDatabase cosmosAsyncDatabase = asyncClient.createDatabaseIfNotExists(this.configuration.getDatabaseId()).block().getDatabase();
+                        CosmosAsyncContainer cosmosAsyncContainer = cosmosAsyncDatabase.createContainerIfNotExists(configuration.getCollectionId(), PARTITION_KEY, 100000).block().getContainer();
```

Comment:
if we are creating the collection, we should allow the throughput to be configurable similar to collectionId and databaseId.

Reply:
it meant to be , had created configuration property for it .It was missed during usage , fixed it .